src/v/cloud_storage/remote_segment.cc

```diff
@@ -256,19 +256,20 @@ ss::future<> remote_segment::do_hydrate() {
           std::move(fparse), std::move(fput));
         bool index_prepared = true;
         if (rparse.failed()) {
+            auto parse_exception = rparse.get_exception();
             vlog(
               _ctxlog.warn,
               "Failed to build a remote_segment index, error: {}",
-              rparse.get_exception());
-            rparse.ignore_ready_future();
+              parse_exception);
             index_prepared = false;
         }
         if (rput.failed()) {
-            vlog(
+            auto put_exception = rput.get_exception();
+            auto vlog(
               _ctxlog.warn,
               "Failed to write a segment file to cache, error: {}",
-              rput.get_exception());
-            rput.get();
+              put_exception);
+            throw put_exception;
```

Comment:
That shouldn't work because `ss::future<>::get_exception` returns `std::exception_ptr`. You can rethrow exception using `std::rethrow_exception` function.

Reply:
Fixed!